pkgs/by-name/we/wemeet/package.nix

```diff
@@ -0,0 +1,250 @@
+      # Comment out `-D WRAP_FORCE_SINK_HARDWARE` to disable the patch that forces wemeet detects sink as hardware sink
+      gcc $CFLAGS -Wall -Wextra -fPIC -shared \
+        "''${openssl_args[@]}" "''${libpulse_args[@]}" \
+        -o libwemeetwrap.so wrap.c -D WRAP_FORCE_SINK_HARDWARE
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      install -Dm755 ./libwemeetwrap.so $out/lib/libwemeetwrap.so
+
+      runHook postInstall
+    '';
+
+    meta = {
+      license = lib.licenses.unfree;
+      maintainers = with lib.maintainers; [ aucub ];
+    };
+  };
+  selectSystem =
+    attrs:
+    attrs.${stdenv.hostPlatform.system}
+      or (throw "wemeet: ${stdenv.hostPlatform.system} is not supported");
+in
+stdenv.mkDerivation {
+  pname = "wemeet";
+  version = "3.19.2.400";
+
+  src = selectSystem {
+    x86_64-linux = fetchurl {
+      url = "https://updatecdn.meeting.qq.com/cos/fb7464ffb18b94a06868265bed984007/TencentMeeting_0300000000_3.19.2.400_x86_64_default.publish.officialwebsite.deb";
+      hash = "sha256-PSGc4urZnoBxtk1cwwz/oeXMwnI02Mv1pN2e9eEf5kE=";
+    };
+    aarch64-linux = fetchurl {
+      url = "https://updatecdn.meeting.qq.com/cos/867a8a2e99a215dcd4f60fe049dbe6cf/TencentMeeting_0300000000_3.19.2.400_arm64_default.publish.officialwebsite.deb";
+      hash = "sha256-avN+PHKKC58lMC5wd0yVLD0Ct7sbb4BtXjovish0ULU=";
+    };
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    makeWrapper
+  ];
+
+  buildInputs = [
+    nss
+    xorg.libX11
+    xorg.libSM
+    xorg.libICE
+    xorg.libXtst
+    desktop-file-utils
+    libpulseaudio
+    libgcrypt
+    dbus
+    systemd
+    udev
+    libGL
+    fontconfig
+    freetype
+    openssl
+    wayland
+    libdrm
+    harfbuzz
+    openldap
+    curl
+    nghttp2
+    libunwind
+    alsa-lib
+    libidn2
+    rtmpdump
+    libpsl
+    libkrb5
+    xkeyboard_config
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    cp -r opt/wemeet $out
+    cp -r usr/* $out
+    rm -f $out/lib/libcurl.so
+    substituteInPlace $out/share/applications/wemeetapp.desktop \
+      --replace-fail "/opt/wemeet/wemeetapp.sh" "wemeetapp" \
+      --replace-fail "/opt/wemeet/wemeet.svg" "wemeet"
+    substituteInPlace $out/bin/qt.conf \
+      --replace-fail "Prefix = ../" "Prefix = $out/lib"
+    cp -r $out/icons $out/share/icons || true
+    install -Dm0644 $out/wemeet.svg $out/share/icons/hicolor/scalable/apps/wemeet.svg
+    ln -s $out/bin/raw/xcast.conf $out/bin/xcast.conf
+    ln -s $out/plugins $out/lib/plugins
+    ln -s $out/resources $out/lib/resources
+    mkdir -p $out/lib/translations
+    ln -s $out/translations/qtwebengine_locales $out/lib/translations/qtwebengine_locales
+
+    runHook postInstall
+  '';
+
+  # set LP_NUM_THREADS limit the number of cores used by rendering
+  # set XDG_SESSION_TYPE; set EGL_PLATFORM; unset WAYLAND_DISPLAY getting border shadows to work
```

Comment:
Isn't that forcing xwayland?

Reply:
> Isn't that forcing xwayland?

It is unfortunately quite common for Tencent's proprietary apps to not support Wayland natively, since they frankly just don't care about ever supporting it. (I think the install script even directly says it will crash under Wayland)